pkg/sql/sem/builtins/builtins.go

```diff
@@ -11254,3 +11303,51 @@ func bitmaskOp(aStr, bStr string, op func(byte, byte) byte) (*tree.DBitArray, er
 
 	return tree.ParseDBitArray(string(buf))
 }
+
+func makeTimestampStatementBuiltinOverload(withTZ bool, withInputTZ bool) tree.Overload {
+	info := "Create timestamp"
+	typs := tree.ParamTypes{{Name: "year", Typ: types.Int}, {Name: "month", Typ: types.Int}, {Name: "day", Typ: types.Int},
+		{Name: "hour", Typ: types.Int}, {Name: "min", Typ: types.Int}, {Name: "sec", Typ: types.Float}}
+	returnTyp := types.Timestamp
+	if withTZ {
+		info += "with time zone from year, month, day, hour, minute and seconds fields (negative years signify BC). If " +
+			"timezone is not specified, the current time zone is used."
+		returnTyp = types.TimestampTZ
+		if withInputTZ {
```

Comment:
nit: Extract this `if` statement from being nested within `withTZ` or perform an assertion that `withInputTZ` is true iff `withTZ` is true. The current structure could cause a lot of confusion if someone happened to call this function with `(false, true)`, if I'm reading it correctly.